scaleway/resource_instance_private_nic_test.go

```diff
@@ -17,34 +17,29 @@ func init() {
 	})
 }
 
-func testSweepInstancePrivateNic(zone string) error {
-	z, err := scw.ParseZone(zone)
-	if err != nil {
-		return err
-	}
-	scwClient, err := sharedClientForZone(z)
-	if err != nil {
-		return fmt.Errorf("error getting client in sweeper: %s", err)
-	}
-	instanceAPI := instance.NewAPI(scwClient)
-
-	l.Debugf("sweeper: destroying the VPC private networks in %s", zone)
-	listInstancePrivateNICs, err := instanceAPI.ListPrivateNICs(&instance.ListPrivateNICsRequest{}, scw.WithAllPages())
-	if err != nil {
-		return fmt.Errorf("error listing private networks in (%s) in sweeper: %s", zone, err)
-	}
-
-	for _, n := range listInstancePrivateNICs.PrivateNics {
-		err := instanceAPI.DeletePrivateNIC(&instance.DeletePrivateNICRequest{
-			PrivateNicID: n.ID,
-			Zone:         z,
-		})
+func testSweepInstancePrivateNic(_ string) error {
+	return sweepZones(scw.AllZones, func(scwClient *scw.Client, zone scw.Zone) error {
```

Comment:
Does deleting server delete private NIC automatically ? if yes server sweeper is enough

Reply:
Yes deleting a server deletes its private NICs

Reply:
removed the nic sweeper and used server sweeper